File: tests/conformance/fixtures/typescript/07-secret-refs.ts
import { Pipeline, fromEnv, fromFile, fromVault } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('deploy').run('./deploy.sh')
  .secretFrom('db_pass', fromEnv('DB_PASSWORD'))
  .secretFrom('tls_cert', fromFile('/certs/tls.pem'))
  .secretFrom('api_key', fromVault('secret/data/api#key'));

p.emit();

File: tests/conformance/fixtures/go/07-secret-refs.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("deploy").Run("./deploy.sh").
		SecretFrom("db_pass", sykli.FromEnv("DB_PASSWORD")).
		SecretFrom("tls_cert", sykli.FromFile("/certs/tls.pem")).
		SecretFrom("api_key", sykli.FromVault("secret/data/api#key"))

	p.Emit()
}
